base/shared/src/main/scala/scalaz/typeclass/DebugSyntax.scala

```diff
@@ -2,9 +2,25 @@ package scalaz
 package typeclass
 
 import language.experimental.macros
+import language.implicitConversions
 
 trait DebugSyntax {
   implicit final class ToDebugOps[A: Debug](self: A) {
     def debug: String = macro meta.Ops.f_0
   }
+
+  implicit final def debugInterpolator(sc: StringContext): DebugInterpolator.DebugInterpolator =
+    DebugInterpolator.DebugInterpolator(sc)
+}
+
+object DebugInterpolator {
+  final case class Debugs(override val toString: String) extends AnyVal
+
+  object Debugs {
+    implicit def mat[A](x: A)(implicit D: Debug[A]): Debugs = Debugs(D.debug(x))
+  }
+
+  final case class DebugInterpolator(sc: StringContext) extends AnyVal {
```

Comment:
having `class DebugInterpolator` _inside_ `object DebugInterpolator` feels kinda weird.

also, and this is _really_ minor, this being a case class means that `sc"fnord"` will give the error

```scala
scala> sc"fnord"
<console>:13: error: StringContext does not take parameters
       sc"fnord"
       ^
```

rather than

```scala
scala> cs"fnord"
<console>:13: error: value cs is not a member of StringContext
       cs"fnord"
       ^
```

... not like anyone would do that, though, right?

Bonus:

```scala
scala> equals"${StringContext("", "")}"
res16: Boolean = true
```